### PR TITLE
Bug/missing repo check error message

### PR DESCRIPTION
**Pull request by yhaliaw**
Applicable spec: NA

### Overview

Only half the error message from repo check is printed out. Fix this issue.
It was found that stopping the runner application prevent cURL from printing out the full body of response. Anything after a newline is cut off.

The fix is using the `logger` command to print out the content as `logger` functions normally under the scenario.

The integration test is modified to check for the reason of failure, e.g., `commit the job is running on is not signed`, to catch this issue if it pops up again.

### Rationale

The user are unable to see the reason for the repo check failure without this change.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: tests/integration/test_charm_fork_repo.py [hunk 1]
```diff
@@ -188,10 +188,7 @@ async def test_dispatch_workflow_failure(
             in logs
         ):
             assert run.jobs()[0].conclusion == "failure"
-            assert (
-                "Stopping execution of jobs due to repository setup is not compliant with policies"
-                in logs
-            )
```
Comment (cbartz) [inline tests/integration/test_charm_fork_repo.py:194]: I think this message could be valuable and we could still check it.

To make the test goals more explicit, we could also change the docstring (L159) from
`assert: The workflow that was dispatched failed.` to `assert: The workflow that was dispatched failed and the reason is logged.`

**Review by jdkandersson — DISMISSED** [review]
Looks good with the change that Chritopher proposed

File: tests/integration/test_charm_fork_repo.py (hunk 1 above)
Comment (yhaliaw) [reply tests/integration/test_charm_fork_repo.py]: Updated the code

**Review by jdkandersson — APPROVED** [review]

**Review by cbartz — APPROVED** [review]
LGTM

**Pull request merged**